src/app/drive/components/ItemDetailsDialog/ItemDetailsDialog.tsx

```diff
@@ -126,32 +126,37 @@ const ItemDetailsDialog = ({
     modified: string,
     email: string,
   ) {
-    const uuid = item.isFolder ? item.uuid : item.folderUuid;
+    const itemType = item.isFolder ? 'folder' : 'file';
+    const itemUuid = item.uuid;
     const rootPathName = item.view;
 
-    const ancestors = await newStorageService.getFolderAncestors(uuid as string);
+    const ancestors = await newStorageService.getFolderAncestorsV2(itemUuid, itemType);
 
     const getPathName = ancestors.map((ancestor) => getItemPlainName(ancestor as unknown as DriveItemData)).reverse();
-
-    if (item.isFolder) {
-      getPathName.pop();
-    }
-
-    if (item.view === 'Drive') {
-      getPathName.shift();
+    getPathName.shift();
+
+    let location = '/';
+    if (item.view === 'Shared') {
+      location += translate('sideNav.shared');
+      if (item.isFolder) {
+        location += getPathName.length > 0 ? '/' + getPathName.shift() : '';
+      }
+    } else {
+      if (item.isFolder) {
+        getPathName.pop();
+      }
+      location += rootPathName + (getPathName.length > 0 ? '/' + getPathName.join('/') : '');
```

Comment:
Separate this logic into functions for ease of understanding and maintenance

Reply:
@CandelR 
I left a note about it in the PR description (**Additional Notes**). Let me know if it's okay or if I should remove the change.